src/shogun/modelselection/ModelSelectionParameter.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Eleftherios Avramidis
+ */
+
+#ifndef __MODELSELECTIONPARAMETER_H_
+#define __MODELSELECTIONPARAMETER_H_
+
+#include <shogun/lib/config.h>
+
+#include <shogun/base/SGObject.h>
+#include <shogun/base/AnyParameter.h>
+
+namespace shogun
+{
+
```

Comment:
I am not sure I agree with this idea of making a class for modelselection parameters....we will have to discuss a bit here.
But actually, as discussed a few day back in person, let's maybe start with the gradient parameter API refactoring